Bambuddy: cannot update (cannot remove '/opt/bambuddy/archive')

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

Bambuddy

### 📂 What was the exact command used to execute the script?

update (within the LCX)

### ⚙️ What settings are you using?

- [ ] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 🧱 Is this Proxmox host running arm64?

No

### 📈 Which Proxmox version are you on?

pve-manager/8.3.5/dac3aa88bac3f300 (running kernel: 6.8.12-9-pve)

### 📝 Provide a clear and concise description of the issue.

I attempted to update Bambuddy, but the process failed.

It looks like it tries to delete the `archive` directory and cant. This is likely because i have passed through a mount point (from the pve host), so that i can store my archive data on my nas.

<img width="532" height="135" alt="Image" src="https://github.com/user-attachments/assets/b5e2c3e7-5465-4ce2-8828-b5098b5a1775" />

### 🔄 Steps to reproduce the issue.

from within the lxc:

```
> update
```

### ❌ Paste the full error output (if available).

```
    ____                  __              __    __
   / __ )____ _____ ___  / /_  __  ______/ /___/ /_  __
  / __  / __ `/ __ `__ \/ __ \/ / / / __  / __  / / / /
 / /_/ / /_/ / / / / / / /_/ / /_/ / /_/ / /_/ / /_/ /
/_____/\__,_/_/ /_/ /_/_.___/\__,_/\__,_/\__,_/\__, /
                                              /____/

  ✔️  Update available: bambuddy 0.2.4.8 → 1.2.5.2
  ✔️  Stopped Service
  💡  Skipping backup of '/opt/bambuddy/bambutrack.db' (not found)
  ✔️  Backed up data to /opt/bambuddy.backup
  ✔️  Backed up Configuration and Data
  ⏳  Fetching GitHub release: bambuddy (1.2.5.2)find: cannot delete ‘/opt/bambuddy/archive’: Device or resource busy
  ✔️  Deployed: bambuddy (1.2.5.2)
  ⏳  Updating Python DependenciesUsing CPython 3.13.14
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
Resolved 88 packages in 1.58s
      Built pyftpdlib==2.2.0
      Built http-ece==1.2.1
Prepared 88 packages in 4.95s
Installed 88 packages in 59ms
 + aiofiles==25.1.0
 + aioftp==0.28.0
 + aiohappyeyeballs==2.7.1
 + aiohttp==3.14.3
 + aiosignal==1.4.0
 + aiosqlite==0.22.1
 + annotated-doc==0.0.5
 + annotated-types==0.8.0
 + anyio==4.14.2
 + asyncpg==0.31.0
 + asyncssh==2.24.0
 + attrs==26.1.0
 + bcrypt==5.0.0
 + certifi==2026.7.22
 + cffi==2.1.1
 + charset-normalizer==3.5.0
 + click==8.4.2
 + contourpy==1.3.3
 + cryptography==50.0.0
 + curl-cffi==0.16.0
 + cycler==0.12.1
 + defusedxml==0.7.1
 + et-xmlfile==2.0.0
 + fast-simplification==0.2.0
 + fastapi==0.135.4
 + fonttools==4.63.0
 + frozenlist==1.8.0
 + greenlet==3.5.5
 + h11==0.16.0
 + http-ece==1.2.1
 + httpcore==1.0.9
 + httptools==0.8.0
 + httpx==0.28.1
 + idna==3.18
 + iniconfig==2.3.0
 + kiwisolver==1.5.0
 + ldap3==2.9.1
 + lxml==6.1.1
 + matplotlib==3.11.1
 + multidict==6.7.1
 + networkx==3.6.1
 + numpy==2.5.2
 + opencv-python-headless==5.0.0.93
 + openpyxl==3.1.5
 + packaging==26.3
 + paho-mqtt==2.1.0
 + passlib==1.7.4
 + pillow==12.3.0
 + pluggy==1.6.0
 + propcache==0.5.2
 + psutil==7.2.2
 + py-vapid==1.9.4
 + pyasn1==0.6.4
 + pyasynchat==1.0.5
 + pyasyncore==1.0.5
 + pycparser==3.0
 + pydantic==2.13.4
 + pydantic-core==2.46.4
 + pydantic-settings==2.15.0
 + pyftpdlib==2.2.0
 + pygments==2.20.0
 + pyjwt==2.13.0
 + pyopenssl==26.4.0
 + pyotp==2.10.0
 + pyparsing==3.3.2
 + pytest==9.1.1
 + pytest-asyncio==1.4.0
 + python-dateutil==2.9.0.post0
 + python-dotenv==1.2.2
 + python-multipart==0.0.32
 + pywebpush==2.4.0
 + pyyaml==6.0.3
 + qrcode==8.2
 + reportlab==5.0.0
 + requests==2.34.2
 + ruff==0.16.3
 + six==1.17.0
 + sqlalchemy==2.0.52
 + starlette==1.6.0
 + trimesh==5.0.0
 + typing-extensions==4.16.0
 + typing-inspection==0.4.4
 + urllib3==2.7.0
 + uvicorn==0.52.3
 + uvloop==0.22.1
 + watchfiles==1.2.0
 + websockets==17.0.1
 + yarl==1.24.5
  ✔️  Updated Python Dependencies
npm warn deprecated whatwg-encoding@3.1.1: Use @exodus/bytes instead for a more spec-conformant and faster implementation

added 606 packages, and audited 607 packages in 12s

213 packages are looking for funding
  run `npm fund` for details

6 vulnerabilities (1 moderate, 5 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
npm notice
npm notice New major version of npm available! 10.9.8 -> 12.0.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v12.0.2
npm notice To update run: npm install -g npm@12.0.2
npm notice

> frontend@0.0.0 build
> tsc -b && vite build

vite v8.0.16 building client environment for production...
✓ 2861 modules transformed.
computing gzip size...
../static/index.html                     1.83 kB │ gzip:     0.82 kB
../static/assets/index-C_6BSgrK.css    198.26 kB │ gzip:    27.96 kB
../static/assets/index-DPZgvI9N.js   9,241.75 kB │ gzip: 2,493.24 kB

✓ built in 3.99s
[plugin builtin:vite-reporter]
(!) Some chunks are larger than 3000 kB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rolldownOptions.output.codeSplitting to improve chunking: https://rolldown.rs/reference/OutputOptions.codeSplitting
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
  ✔️  Rebuilt Frontend
  ⏳  Restoring datarm: cannot remove '/opt/bambuddy/archive': Device or resource busy
  ✖️  in line 1365: exit code 1 (General error / Operation not permitted): while executing command rm -rf "$path"

  --- Last 20 lines of log ---
[2026-08-15 08:25:56] [INFO] Checking for update: bambuddy
[2026-08-15 08:25:56] [OK] Update available: bambuddy 0.2.4.8 → 1.2.5.2
[2026-08-15 08:25:56] [INFO] Stopping Service
[2026-08-15 08:25:57] [OK] Stopped Service
[2026-08-15 08:25:57] [INFO] Backing up Configuration and Data
[2026-08-15 08:25:57] [INFO] Backing up data
[2026-08-15 08:25:57] [WARN] Skipping backup of '/opt/bambuddy/bambutrack.db' (not found)
[2026-08-15 08:25:57] [OK] Backed up data to /opt/bambuddy.backup
[2026-08-15 08:25:57] [OK] Backed up Configuration and Data
[2026-08-15 08:25:57] [INFO] Fetching GitHub release: bambuddy (1.2.5.2)
[2026-08-15 08:26:02] [OK] Deployed: bambuddy (1.2.5.2)
[2026-08-15 08:26:02] [INFO] Updating Python Dependencies
[2026-08-15 08:26:09] [OK] Updated Python Dependencies
[2026-08-15 08:26:09] [INFO] Rebuilding Frontend
[2026-08-15 08:27:23] [OK] Rebuilt Frontend
[2026-08-15 08:27:23] [INFO] Restoring data
[2026-08-15 08:27:24] [ERROR] in line 1365: exit code 1 (General error / Operation not permitted): while executing command rm -rf "$path"
  -----------------------------------
```

### 🖼️ Additional context (optional).

_No response_

--- a/misc/tools.func
+++ b/misc/tools.func
@@ -1336,6 +1336,10 @@ create_backup() {
       msg_warn "Skipping backup of '${path}' (not found)"
       continue
     fi
+    if mountpoint -q "$path" 2>/dev/null; then
+      msg_warn "Skipping backup of '${path}' (is a mount point)"
+      continue
+    fi
     dest="${store}/files${path}"
     if ! mkdir -p "$(dirname "$dest")" || ! cp -a "$path" "$dest"; then
       msg_error "Backup of '${path}' failed - aborting update"
@@ -2538,7 +2542,7 @@ _deploy_source_tarball() {
 
   mkdir -p "$target"
   if [[ "${CLEAN_INSTALL:-0}" == "1" ]]; then
-    find "${target:?}" -mindepth 1 -delete
+    find "${target:?}" -mindepth 1 \( -type d -exec mountpoint -q {} \; -prune \) -o -delete
   fi
 
   tar --no-same-owner -xzf "$tarball" -C "$workdir" || {
@@ -2627,7 +2631,7 @@ _deploy_unpacked_archive() {
   # was truncated, the archive was unreadable, or it unpacked to nothing.
   mkdir -p "$target"
   if [[ "${CLEAN_INSTALL:-0}" == "1" ]]; then
-    find "${target:?}" -mindepth 1 -delete
+    find "${target:?}" -mindepth 1 \( -type d -exec mountpoint -q {} \; -prune \) -o -delete
   fi
 
   if ! cp -r "$source_dir"/* "$target/"; then
@@ -9368,7 +9372,7 @@ fetch_and_deploy_from_url() {
   mkdir -p "$directory"
 
   if [[ "${CLEAN_INSTALL:-0}" == "1" ]]; then
-    find "${directory:?}" -mindepth 1 -delete
+    find "${directory:?}" -mindepth 1 \( -type d -exec mountpoint -q {} \; -prune \) -o -delete
   fi
 
   local unpack_tmp

--- a/misc/tools.func
+++ b/misc/tools.func
@@ -2528,9 +2528,9 @@ _download_source_tarball() {
 # directory). Extracts <tarball_path> into <workdir>, then copies the contents
 # of that top-level directory into <target>.
 #
-#   - Honors CLEAN_INSTALL=1 (wipes <target> first, dotfiles included — back
-#     up config dotfiles like .env via create_backup and call restore_backup
-#     BEFORE any build step that sources them).
+#   - Honors CLEAN_INSTALL=1 (wipes <target> first, dotfiles included, mount
+#     points preserved — back up config dotfiles like .env via create_backup
+#     and call restore_backup BEFORE any build step that sources them).
 #   - Does NOT own <workdir>: the caller creates it and is responsible for its
 #     cleanup (typically via a RETURN trap on its tmpdir).
 #   - cp failures are non-fatal here, matching the previous inline behavior.
@@ -2568,9 +2568,9 @@ _deploy_source_tarball() {
 # a single top-level directory, that directory is stripped (its contents land
 # directly in <target>); otherwise the archive contents are copied as-is.
 #
-#   - Honors CLEAN_INSTALL=1 (wipes <target> first, dotfiles included — back
-#     up config dotfiles like .env via create_backup and call restore_backup
-#     BEFORE any build step that sources them).
+#   - Honors CLEAN_INSTALL=1 (wipes <target> first, dotfiles included, mount
+#     points preserved — back up config dotfiles like .env via create_backup
+#     and call restore_backup BEFORE any build step that sources them).
 #   - Does NOT own <workdir>: the caller creates it and cleans it up.
 #
 # Returns: 0 on success, 65 on unsupported format, 251 on extraction failure,

